src/renderer/webgl/shaders/SpinnerShader2.mjs

```diff
@@ -0,0 +1,184 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import DefaultShader from "./DefaultShader.mjs";
+import StageUtils from "../../../tree/StageUtils.mjs";
+
+export default class SpinnerShader2 extends DefaultShader {
+    constructor(context) {
+        super(context);
+        this._period = 1;
+        this._stroke = 0;
+        this._showDot = true;
+        this._clockwise = true;
+        this._bc = 0xff000000;
+        this._normalizedBC = this._getNormalizedColor(this._bc);
+        this._c = 0xffffffff;
+        this._normalizedC = this._getNormalizedColor(this._c);
+    }
+
+    set radius(v) {
+        if(v === 0) {
+            v = 1;
+        }
+        this._radius = v;
+    }
+
+    set stroke(value) {
+        this._stroke = Math.abs(value);
+    }
+
+    get stroke() {
+        return this._stroke;
+    }
+
+    set time(ts) {
```

Comment:
This shouldn't be part of API, it's very hard to make good use of it. It should be either removed or replaced by `angle` to make initial angle configuration possible.

Reply:
No because it is not related to the angle

Reply:
That setter is so you can synchronize your spinners rotation with other spinners

Reply:
Wouldn't it be better for it to have constant time baseline set as default so that all spinners are in sync by default?